qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py

```diff
@@ -75,8 +77,11 @@ def __init__(
         """
         super().__init__(neural_network, loss, optimizer, warm_start, initial_point)
         self._one_hot = one_hot
+        self._target_encoder = None  # encodes the target data if categorical
```

Comment:
```suggestion
        self._target_encoder: Union[OneHotEncoder, LabelEncoder] = None  # encodes target data if categorical
```
Lets see if this typehint helps mypy; otherwise I think its not picking up on the type and giving errors about None not having fit or transform. What might be better than this though, if it works, would be to say its a `TransformerMixin` type and instead of doing `fit` then `transform` in the code below use the `fit_transform` defined by this, which seems to be a convenience method for calling the two in sequence. The encoders implement that Mixin so it seemed to be the better solution.

Reply:
I committed this, so it got marked resolved, but unresolved it so the comment is readily visible about TransformerMixin. The typehint change (after I fixed it up!) did sort things out though for mypy.

Reply:
My bad this can't be done. As fit is always called but transform is conditional!